tests/integrationv2/test_key_update.py

```diff
@@ -0,0 +1,65 @@
+import copy
+import pytest
+import time
+
+from configuration import available_ports, Ciphers, ALL_TEST_CURVES, ALL_TEST_CERTS, PROVIDERS, PROTOCOLS
+from common import ProviderOptions, Protocols, data_bytes
+from fixtures import managed_process
+from providers import Provider, S2N, OpenSSL
+from utils import invalid_test_parameters, get_parameter_name, get_expected_s2n_version
+
+
+
+@pytest.mark.uncollect_if(func=invalid_test_parameters)
+# These are the only ciphers that utilize the key update feature
+@pytest.mark.parametrize("cipher", [Ciphers.AES128_GCM_SHA256, Ciphers.AES256_GCM_SHA384])
+@pytest.mark.parametrize("curve", ALL_TEST_CURVES)
+@pytest.mark.parametrize("certificate", ALL_TEST_CERTS, ids=get_parameter_name)
+def test_s2n_server_key_update(managed_process, cipher, curve, certificate):
+    host = "localhost"
+    port = next(available_ports)
+    
+    update_requested = b"K"
+    server_data = data_bytes(10)
+    client_data = data_bytes(10)
+
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        host=host,
+        port=port,
+        cipher=cipher,
+        curve=curve,
+        data_to_send=update_requested + client_data,
```

Comment:
Yeah I originally wanted to add the data in a queue. The queue takes away the issue of chunk_size entirely, because we're just writing whatever gets popped off the list. I wasn't sure which approach would be more pythonic, utilizing the chunk size to cut the message to the right length, or popping items off a queue.